client/client.go

```diff
@@ -89,4 +89,36 @@ type OpAMPClient interface {
 	// May be called before or after Start().
 	// May be also called from OnMessage handler.
 	RequestConnectionSettings(request *protobufs.ConnectionSettingsRequest) error
+
+	// SetCustomCapabilities modifies the set of customCapabilities supported by the client.
+	// The new customCapabilities will be sent with the next message to the server. SHOULD
```

Comment:
Do we want to a "SHOULD" here? If someone is not using custom capabilities then they don't need to call this. Perhaps we can phrase a bit differently "If custom capabilities are used SHOULD be called before Start()". What do you think?